eng/pipelines/templates/scripts/assemble-dotnet-standalone-exe.ps1

```diff
@@ -0,0 +1,92 @@
+<#
+.SYNOPSIS 
+Publishes a standalone dotnet executable to an artifact staging directory.
+
+.DESCRIPTION
+Assembles a standalone executable and places it within the given staging directory. This script takes care of any additional minutae that is required to
+enable a usable binary down the line after signing or notarization.
+
+.PARAMETER Rid
+The target platform. Takes the form of "osx-x64", "win-arm64", "linux-x64", etc. A full list is available here: https://learn.microsoft.com/en-us/dotnet/core/rid-catalog
+
+.PARAMETER ArtifactStagingDirectory
+The root directory which will receive the compressed standalone executable. 
+
+.PARAMETER Target 
+The targeted folder that should be built and assembled into a standalone executable.
+
+.PARAMETER Framework
+The targeted .NET framework. Defaults to "net6.0."
+
+#>
+param(
+   [Parameter(mandatory=$true)]
+   [string] $Rid,
+   [Parameter(mandatory=$true)]
+   [string] $Target,
+   [Parameter(mandatory=$true)]
+   [string] $ArtifactStagingDirectory,
+   [Parameter(mandatory=$false)]
+   [string] $Framework = "net6.0"
+)
+
+# resolves to <artifactfolder>/win-x64
+$destinationArtifactFolder = Join-Path $($ArtifactStagingDirectory) $($Rid)
+
+# resolves to <artifactfolder>/win-x64/test-proxy-standalone-win-x64 (.zip or .tar.gz will be added as appropriate for platform)
+$destinationPathSegment = Join-Path $destinationArtifactFolder "$((Split-Path -Leaf "$($Target)"))-standalone-$Rid"
+
+# resolves to tools/test-proxy/win-x64
+$outputPath = Join-Path $($Target) $($Rid)
+
+# ensure the destination artifact directory exists
+if (!(Test-Path $destinationArtifactFolder)){
+   New-Item -Force -Path $destinationArtifactFolder -ItemType directory
+}
+
+Write-Host "dotnet publish -f $($Framework) -c Release -r $($Rid) -p:PublishSingleFile=true --self-contained --output $outputPath $Target"
+dotnet publish -f $($Framework) -c Release -r $($Rid) -p:PublishSingleFile=true --self-contained --output $outputPath $Target
```

Comment:
A tad more readable without the unnecessary string wrappers since you aren't referencing any properties:

```suggestion
$destinationArtifactFolder = Join-Path $ArtifactStagingDirectory $Rid

# resolves to <artifactfolder>/win-x64/test-proxy-standalone-win-x64 (.zip or .tar.gz will be added as appropriate for platform)
$destinationPathSegment = Join-Path $destinationArtifactFolder "$(Split-Path -Leaf "$Target")-standalone-$Rid"

# resolves to tools/test-proxy/win-x64
$outputPath = Join-Path $Target $Rid

# ensure the destination artifact directory exists
if (!(Test-Path $destinationArtifactFolder)){
   New-Item -Force -Path $destinationArtifactFolder -ItemType directory
}

Write-Host "dotnet publish -f $Framework -c Release -r $Rid -p:PublishSingleFile=true --self-contained --output $outputPath $Target"
dotnet publish -f $Framework -c Release -r $Rid -p:PublishSingleFile=true --self-contained --output $outputPath $Target
```